lib/_emerge/depgraph.py

```diff
@@ -5205,6 +5206,22 @@ def _resolve(self, myfavorites):
                         if not package_is_installed:
                             continue
 
+                    # If we're emerging @selected or @world, we want to loudly warn about
+                    # no ebuilds being available for packages (bug #911180).
+                    if (
+                        pkg
+                        and pkg.installed
+                        and pkg.operation == "nomerge"
+                        and isinstance(arg, SetArg)
+                        and arg.name in ("selected, world")
+                        and not self._replace_installed_atom(pkg, False)
```

Comment:
@zmedico Does this seem okay to you? It feels like a good fit and I've added a parameter to skip some of the extra matching it tries to do, as I don't think we need it here.

This feels a bit better than reinventing another check to look through repositories for an existing match, and it also matches the check for the thing we really care about (subslot conflicts where something can't get rebuilt).

WDYT?

Reply:
@zmedico ping when you get a chance but this one isn't really urgent (and you've done enough for me today anyway 😄)